src/pages/workspaces/workspace/tools/WorkflowView.js

```diff
@@ -23,12 +27,38 @@ const tableColumns = [
 
 const tabs = ['Inputs', 'Outputs', 'WDL']
 
+const styleForOptional = (optional, text) =>
+  span({
+    style: {
+      fontWeight: !optional && 500,
+      fontStyle: optional && 'italic',
+      overflow: 'hidden', textOverflow: 'ellipsis'
+    }
+  }, [text])
+
+const miniMessage = text =>
+  span({ style: { fontWeight: 500, fontSize: '75%', marginRight: '1rem', textTransform: 'uppercase' } }, [text])
+
+const extractTaskAndVariable = list => {
+  return _.map(list, entry => {
+    const [task, variable] = _.takeRight(_.split(entry.name, '.'), 2)
+    return _.merge(entry, { task, variable })
+  })
+}
+
 
 class WorkflowView extends Component {
   constructor(props) {
     super(props)
 
-    this.state = { selectedTab: 'Inputs', loadedWdl: false }
+    this.state = {
+      selectedTab: 'Inputs', loadedWdl: false, untouched: true,
+      modifiedAttributes: { inputs: {}, outputs: {} }
+    }
+
+    const { workspaceNamespace, workspaceName, workflowNamespace, workflowName } = props
+    this.rawlsMethodConfig = Rawls.workspace(workspaceNamespace, workspaceName).methodConfig(workflowNamespace, workflowName)
+    this.workspaceId = { namespace: workspaceNamespace, name: workspaceName }
```

Comment:
This is probably not a good idea, because it ends up caching the initial props. If they change in the future, this could break in subtle ways. If we're really trying to avoid typing these out several times, then a couple helper functions would be a better option.